dotnet/RPCClient/RPCClient.cs

```diff
@@ -11,37 +11,40 @@ class RPCClient
     private IConnection connection;
     private IModel channel;
     private string replyQueueName;
-    private QueueingBasicConsumer consumer;
+    private EventingBasicConsumer consumer;
 
     public RPCClient()
     {
         var factory = new ConnectionFactory() { HostName = "localhost" };
         connection = factory.CreateConnection();
         channel = connection.CreateModel();
         replyQueueName = channel.QueueDeclare().QueueName;
-        consumer = new QueueingBasicConsumer(channel);
+        consumer = new EventingBasicConsumer(channel);
         channel.BasicConsume(queue: replyQueueName, autoAck: true, consumer: consumer);
     }
 
-    public string Call(string message)
-    {
-        var corrId = Guid.NewGuid().ToString();
-        var props = channel.CreateBasicProperties();
-        props.ReplyTo = replyQueueName;
-        props.CorrelationId = corrId;
+        public string Call(string message)
+        {
+            var corrId = Guid.NewGuid().ToString();
+            var props = channel.CreateBasicProperties();
+            string response = null;
 
-        var messageBytes = Encoding.UTF8.GetBytes(message);
-        channel.BasicPublish(exchange: "", routingKey: "rpc_queue", basicProperties: props, body: messageBytes);
+            props.ReplyTo = replyQueueName;
+            props.CorrelationId = corrId;
 
-        while(true)
-        {
-            var ea = (BasicDeliverEventArgs)consumer.Queue.Dequeue();
-            if(ea.BasicProperties.CorrelationId == corrId)
+            var messageBytes = Encoding.UTF8.GetBytes(message);
+            channel.BasicPublish(exchange: "", routingKey: "rpc_queue", basicProperties: props, body: messageBytes);
+
+            consumer.Received += (model, ea) =>
```

Comment:
This is not an equivalent version: `Dequeue` in the original version will wait until a message is available. This code won't and may return a `null` response. Use a manual reset event or similar to synchronise.